pagefaultgames/pokerogue
# [Bug] Adjusts bug bite/pluck to remove berry pouch preservation on opponent berries

## What are the changes?

Currently, bug bite/pluck has a chance to preserve the berry in an opponents inventory if the user has berry pouches, potentially creating a nearly endless supply. With these changes, bug bite/pluck will no longer coincide with berry pouch functionality.

## Why am I doing these changes?

Bug bite/pluck has a chance to preserve the berry in an opponents inventory if the user has berry pouches, potentially creating a nearly endless supply.

#1315

Balances bug bite/pluck with berry pouch better to avoid infinite supplies of berries.

## What did change?

Removes the StealEatBerryAttr class extension of EatBerryAttr and removes berry pouch preservation when stealing an opponents berry.

Before:
StealEatBerryAttr extended EatBerryAttr and the super class (EatBerryAttr) was called to eat the berry, gain its effects, and check for berry pouch preservation.
After:
StealEatBerryAttr extends only MoveEffectAttr and no longer calls the super class to eat the berry and no longer checks for berry pouch preservation.


## How to test the changes?

Tested using the `src/overrides.ts` file. 
export const STARTING_HELD_ITEMS_OVERRIDE: Array<ModifierOverride> = [{name: "BERRY", count: 3, type: BerryType.SITRUS}, {name: "BERRY_POUCH", count: 3}];
export const OPP_HELD_ITEMS_OVERRIDE: Array<ModifierOverride> = [{name: "BERRY", count: 3, type: BerryType.SITRUS}];

Recommend testing with no berry pouches vs. three berry pouches to see the effect.

## Checklist
